src/components/Widgets/DateControl.js

```diff
@@ -1,18 +1,9 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import DateTimeControl from './DateTimeControl';
 import DateTime from 'react-datetime';
 
-export default class DateControl extends React.Component {
-  componentDidMount() {
-    if (!this.props.value) {
-      this.props.onChange(new Date());
-    }
-  }
-
-  handleChange = (datetime) => {
-    this.props.onChange(datetime);
-  };
-
+export default class DateControl extends DateTimeControl {
```

Comment:
I understand why this makes sense, but extending individual components is something we consider an anti-pattern.

Reply:
Understood - I did a bigger rework and rather than extending it now delegates to the other component instead; I'm hoping this is an acceptable alternative?